homework/index.js

```diff
@@ -29,12 +31,115 @@
     return elem;
   }
 
-  function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+  function addCont(login, img, contributeNumber, parent, link) {
+    const tr = createAndAppend('tr', parent);
+    const imageColumn = createAndAppend('th', tr);
+    createAndAppend('img', imageColumn, {
+      src: img,
+      class: 'contributorImg',
+    });
+    const name = createAndAppend('td', tr);
+    createAndAppend('a', name, {
+      href: link,
+      target: '_blank',
+      text: login,
+    });
+    const contributeNamePlace = createAndAppend('td', tr);
+    createAndAppend('span', contributeNamePlace, {
+      text: contributeNumber,
+      class: 'contributeNumber',
+    });
+  }
+
+  function addRow(property, key, parent) {
+    const tr = createAndAppend('tr', parent);
+    createAndAppend('th', tr, {
+      text: `${property} :`,
+    });
+    createAndAppend('td', tr, { text: key });
+    return tr;
+  }
+
+  let optionNumber = 0;
+  function madeOption(repo, selectButton) {
+    createAndAppend('option', selectButton, {
+      text: repo.name,
+      value: optionNumber,
+    });
+    optionNumber++;
+  }
+
+  function renderContributorDetails(conts, contributorsContainer) {
+    const table = createAndAppend('table', contributorsContainer);
+    conts.forEach(cont => {
+      addCont(
+        cont.login,
+        cont.avatar_url,
+        cont.contributions,
+        table,
+        cont.html_url,
+      );
+    });
+  }
+
+  function renderRepoDetails(repo, div) {
+    const table = createAndAppend('table', div);
+    const firstRow = addRow('Repository', '', table);
+    createAndAppend('a', firstRow.lastChild, {
+      href: repo.html_url,
+      target: '_blank',
+      text: repo.name,
+    });
+    addRow('Description', repo.description, table);
+    addRow('Forks', repo.forks, table);
+    addRow('Updated', repo.updated_at, table);
+  }
+
+  function fetchContributors(url, contributorsContainer) {
+    const promiseContributor = fetchJSON(url);
+    promiseContributor.then((conts, err) => {
+      if (err) {
```

Comment:
Error handling with promises doesn't work this way. Please review again how to `.catch()` errors when using promises.